File: todo-frontend/src/app/models/task.ts
export interface Task {
  id?: number;
  title: string;
  description?: string;
  category: 'travail' | 'étude' | 'personnel';
  priority: 'faible' | 'moyenne' | 'haute';
  dueDate?: string; // تاريخ الاستحقاق (YYYY-MM-DD)
  status: 'Pending' | 'Done'; // استخدام status بدلاً من completed
  user?: any; // المستخدم المرتبط
  createdAt?: string;
  updatedAt?: string;
}

export interface TaskCreateDto {
  title: string;
  description?: string;
  category: string;
  priority: string;
  dueDate?: string;
}

// ثوابت للتصنيفات والأولويات
export const CATEGORIES = {
  TRAVAIL: 'travail' as Task['category'],
  ETUDE: 'étude' as Task['category'],
  PERSONNEL: 'personnel' as Task['category']
} as const;

export const PRIORITIES = {
  FAIBLE: 'faible' as Task['priority'],
  MOYENNE: 'moyenne' as Task['priority'],
  HAUTE: 'haute' as Task['priority']
} as const;

export const STATUS = {
  PENDING: 'Pending' as Task['status'],
  DONE: 'Done' as Task['status']
} as const;

// خيارات العرض
export const CATEGORY_OPTIONS = [
  { value: CATEGORIES.TRAVAIL, label: 'Travail', icon: '💼', color: '#4299e1' },
  { value: CATEGORIES.ETUDE, label: 'Étude', icon: '📚', color: '#ed8936' },
  { value: CATEGORIES.PERSONNEL, label: 'Personnel', icon: '🏠', color: '#48bb78' }
];

export const PRIORITY_OPTIONS = [
  { value: PRIORITIES.FAIBLE, label: 'Faible', color: '#48bb78', bgColor: '#c6f6d5' },
  { value: PRIORITIES.MOYENNE, label: 'Moyenne', color: '#ed8936', bgColor: '#feebc8' },
  { value: PRIORITIES.HAUTE, label: 'Haute', color: '#f56565', bgColor: '#fed7d7' }
];

export const STATUS_OPTIONS = [
  { value: STATUS.PENDING, label: 'En attente', icon: '⏳', color: '#a0aec0' },
  { value: STATUS.DONE, label: 'Terminée', icon: '✅', color: '#48bb78' }
];

// دالة مساعدة للتحويل
export function isTaskCompleted(task: Task): boolean {
  return task.status === 'Done';
}

export function mapToTaskCreateDto(task: Partial<Task>): TaskCreateDto {
  return {
    title: task.title || '',
    description: task.description,
    category: task.category || CATEGORIES.TRAVAIL,
    priority: task.priority || PRIORITIES.MOYENNE,
    dueDate: task.dueDate
  };
}

File: backend/todo-backend/src/main/java/com/todo/todo_backend/model/Task.java
package com.todo.todo_backend.model;

import com.fasterxml.jackson.annotation.JsonBackReference;   // ← أضف هذا الـ import
import jakarta.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;


@Entity
@Table(name = "tasks")
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    private String description;

    @Column(nullable = false)
    private String category; // travail / étude / personnel

    @Column(nullable = false)
    private String priority; // faible / moyenne / haute

    private LocalDate dueDate;

    private String status = "Pending"; // Pending / Done

    private LocalDateTime createdAt = LocalDateTime.now();

    private LocalDateTime updatedAt = LocalDateTime.now();

    // العلاقة مع المستخدم
    @JsonBackReference
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    // ======== Constructors ========
    public Task() {}

    public Task(String title, String category, String priority, User user) {
        this.title = title;
        this.category = category;
        this.priority = priority;
        this.user = user;
    }

    // ======== Getters & Setters ========
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getTitle() { return title; }
    public void setTitle(String title) { 
        this.title = title; 
        this.updatedAt = LocalDateTime.now();
    }

    public String getDescription() { return description; }
    public void setDescription(String description) { 
        this.description = description; 
        this.updatedAt = LocalDateTime.now();
    }

    public String getCategory() { return category; }
    public void setCategory(String category) { 
        this.category = category; 
        this.updatedAt = LocalDateTime.now();
    }

    public String getPriority() { return priority; }
    public void setPriority(String priority) { 
        this.priority = priority; 
        this.updatedAt = LocalDateTime.now();
    }

    public LocalDate getDueDate() { return dueDate; }
    public void setDueDate(LocalDate dueDate) { 
        this.dueDate = dueDate; 
        this.updatedAt = LocalDateTime.now();
    }

    public String getStatus() { return status; }
    public void setStatus(String status) { 
        this.status = status; 
        this.updatedAt = LocalDateTime.now();
    }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }

    public User getUser() { return user; }
    public void setUser(User user) { this.user = user; }
}